stdlib/gettext.pyi

```diff
@@ -67,9 +68,14 @@ else:
         "ngettext",
     ]
 
+class _TranslationsReader(Protocol):
+    def read(self) -> bytes: ...
+    # optional:
+    # name: str
```

Comment:
I didn't use `SupportsRead` deliberately:

* The `__length` argument to `read()` is not required.
* To document the optional `name` field. We could add it if something like python/typing#601 gets implemented.